src/sentry/integrations/msteams/link_identity.py

```diff
@@ -33,7 +34,13 @@ class MsTeamsLinkIdentityView(BaseView):
     @transaction_start("MsTeamsLinkIdentityView")
     @never_cache
     def handle(self, request, signed_params):
-        params = unsign(signed_params)
+        try:
+            params = unsign(signed_params)
+        except (SignatureExpired, BadSignature):
```

Comment:
👍

Reply:
Should we catch `BadSignature` the same way though? Re-linking might not fix that error.

Reply:
@scefali hmm how do we get a bad signature? It should be something wrong with the link that provides the hash. Getting a fresh link should solve that problem.

Reply:
makes sense to me then